File: frontend/src/types/sale.ts
export type PaymentMethod = 'CASH' | 'CREDIT_CARD' | 'DEBIT_CARD' | 'BANK_TRANSFER' | 'FINANCING'
export type SaleStatus = 'PENDING' | 'COMPLETED' | 'CANCELLED' | 'REFUNDED'

export interface SaleItemRequest {
  vehicleId: string
  discount?: number
}

export interface SaleRequest {
  customerId: string
  paymentMethod: PaymentMethod
  items: SaleItemRequest[]
  notes?: string
  installments?: number
  downPayment?: number
}

export interface SaleItemResponse {
  id: number
  vehicleId: string
  vehicleModel: string
  unitPrice: number
  discount: number
}

export interface PaymentResponse {
  id: string
  method: PaymentMethod
  amount: number
  installments: number | null
  installmentAmount: number | null
  paymentDate: string
}

export interface SaleResponse {
  id: string
  customerId: string
  customerName: string
  employeeId: string
  employeeName: string
  totalAmount: number
  paymentMethod: PaymentMethod
  status: SaleStatus
  saleDate: string
  notes: string | null
  items: SaleItemResponse[]
  payments: PaymentResponse[]
  createdAt: string
}

export interface SaleStatusRequest {
  status: SaleStatus
}

File: src/main/java/com/concesionario/entity/Sale.java
package com.concesionario.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.*;

@Entity
@Table(name = "sales")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Sale extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(name = "total_amount", nullable = false, precision = 14, scale = 2)
    private BigDecimal totalAmount;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method", nullable = false, length = 30)
    private PaymentMethod paymentMethod;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SaleStatus status = SaleStatus.PENDING;

    @Column(name = "sale_date", nullable = false)
    @Builder.Default
    private LocalDateTime saleDate = LocalDateTime.now();

    @Column(columnDefinition = "TEXT")
    private String notes;

    @OneToMany(mappedBy = "sale", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SaleItem> items = new ArrayList<>();

    @OneToMany(mappedBy = "sale", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Payment> payments = new ArrayList<>();

    public enum PaymentMethod { CASH, CREDIT_CARD, DEBIT_CARD, BANK_TRANSFER, FINANCING }
    public enum SaleStatus { PENDING, COMPLETED, CANCELLED, REFUNDED }
}

// =============================================
// ---- SaleItem.java ----
